code/datums/elements/_element.dm

```diff
@@ -75,3 +75,28 @@
 		ele.Detach(arglist(arguments))
 	else
 		ele.Detach(src)
+
+/**
+ * Used to manage (typically non_bespoke) elements with multiple sources through traits
+ * so we don't have to make them a components again.
+ * The element will be later removed once all trait sources are gone, there's no need of a
+ * "RemoveElementTrait" counterpart.
+ */
+/datum/proc/AddElementTrait(trait, source, datum/element/eletype, ...)
+	if(!ispath(eletype, /datum/element))
+		CRASH("AddElementTrait called, but [eletype] is not of a /datum/element path")
+	var/already_had_trait = HAS_TRAIT(src, trait)
+	ADD_TRAIT(src, trait, source)
+	if(already_had_trait)
+		return
+	var/list/arguments = list(eletype)
+	if(length(args) > 3)
```

Comment:
magic number, what does this mean

Reply:
It really isn't all that magical, just look at the number of fixed args. I'll give it a comment to explain though.